tests/test-mongoc-collection.c

```diff
@@ -637,6 +639,53 @@ test_index_geo (void)
    mongoc_client_destroy(client);
 }
 
+static const char *
+storage_engine (mongoc_client_t *client)
+{
+   uint32_t len;
+   const uint8_t *data;
+   bson_iter_t iter;
+   bson_iter_t parsed_iter;
+   bson_iter_t storage_iter;
+   bson_error_t error;
+   bson_t cmd = BSON_INITIALIZER;
+   bson_t reply;
+   bson_t parsed;
+   bson_t storage;
+   bool r;
+   const char *engine;
+   /* NOTE: this default will change eventually */
+   const char *unspecified = "mmapv1";
+
+   /* from getCmdLineOpts, if engine was specified, result looks like:
+      { "parsed" : { "storage" : { "engine" : "wiredtiger" }, ... }, ... } */
+   BSON_APPEND_INT32 (&cmd, "getCmdLineOpts", 1);
+   r = mongoc_client_command_simple(client, "admin", &cmd, NULL, &reply, &error);
+   ASSERT (r);
+
+   ASSERT (bson_iter_init_find (&iter, &reply, "parsed"));
```

Comment:
We can eliminate some code and variables in this method by using a dotted argument to bson_iter_init_find:

```c
if( bson_iter_init_find(&iter, &reply, "parsed.storage.engine") ) {
    /// found
}
```